operator_ui/src/components/JobRuns/TaskExpansionPanel.tsx

```diff
@@ -25,7 +25,7 @@ interface IItemProps extends WithStyles<typeof fontStyles> {
   colATitle: string
   colAValue: string
   colBTitle: string
-  colBValue?: string
+  colBValue: string | null
```

Comment:
Was this causing type error? We had a discussion a few weeks ago where we're going to try and use `foo?` instead of `null` as a standard. I think some of the DB types still return `null` so is it possible for them to return `undefined` instead?

Reply:
Yeah this was throwing a type error. Would be happy to switch it over to undefined. Does the DB actually return an undefined key or is it null? We should make it match exactly what the underlying type is

Reply:
My first instinct was to keep it the same as the db also. But I found that a lot of libraries only annotate their function args with `?`. So you have to end up running code with extra guards like `item.myAttribute === null ? someLib(undefined) : someLib(item.myAttribute)`. Any thoughts on how to avoid that?

Reply:
This is for the TS repo itself but we thought we'd try and follow along https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined

Reply:
We would want to keep it exactly what the underlying type is. E.g if the underlying field is null, then we should make the type of it null. Same goes for undefined. The reasoning is: if you type something as 'undefined' but it's actually 'null', you can get unintended behaviour from strict equality checks since they have to match the exact type. `null === undefined` is false

Reply:
So if we want to eliminate the extra guards do you think we'd need to be submitting upstream PR's and use the fork or a monkey patch?